public/scripts/experiments/estimation/estimation.js

```diff
@@ -0,0 +1,465 @@
+        this.input_count_array = new Array(this.curr_conditions_constants[0].trials_per_round).fill(0);
+        this.is_practice = false;
+    }
+
+    /**
+     * Generates a Estimation object for use in the JsPsych timeline.
+     *
+     * @param  block_type {string}     "test" or "practice"
+     * @return trial {object}
+     */
+    generate_trial(block_type) {
+        if ((block_type !== "test") && (block_type !== "practice")) {
+            throw Error(block_type + " is not supported.")
+        }
+
+        // Initialize a variable for this so it is usable inside on_start
+        var estimation_exp = this;
+        var address = location.protocol + "//" + location.hostname + ":" + location.port + "/estimation_trial";
+
+        let trial = {
+            type:'external-html-keyboard-response',
+            url: address,
+            choices: [32, 81],  // 32 = spacebar, 81 = q (exit button for debugging)
+            execute_script: true,
+            response_ends_trial: true,
+            data: {
+                round_num: 0,
+                estimated_size: -1,
+                adjustments: [], // array of numbers representing the adjustments made to the shape
+                is_ref_left: false, // is the reference shape on the left
+                sub_condition_index: 0,
+                block_type: block_type
+            },
+            on_start: function(trial) {
+                console.log("====================on_start=======================");
+                // Set the constants to be used:
+                trial.data.sub_condition_index = estimation_exp.curr_condition_index;
+                trial.data.round_num = estimation_exp.curr_round_num;
+                trial.data = Object.assign(estimation_exp.curr_conditions_constants[estimation_exp.curr_condition_index],
+                    trial.data);
+                estimation_exp.curr_trial_data = trial.data;
+                // Save trial data for practice so can calculate exclusion criteria
+                if (trial.data.run_type === "practice") {
+                    estimation_exp.practice_trial_data[index].push(trial.data);
+                }
+                // console.log(JSON.stringify(trial.data));
+                console.log(JSON.stringify(trial));
+            },
+            on_finish: function(data) { // NOTE: on_finish takes in data var
+                // save data here
+                console.log("====================on_finish=======================");
+                let curr_trail_data = JSON.parse(JSON.stringify((data)));
+                estimation_exp.results.push(curr_trail_data);
+                estimation_exp.update_curr_round_number(data);
+                estimation_exp.update_curr_cond_idx(data);
+                estimation_exp.update_input_array(data);
+                // console.log("RESULTS: " + JSON.stringify(estimation_exp.results));
+            }
+        };
+        return trial;
+    }
+
+    update_input_array(data) {
```

Comment:
Add method comments (+ for the methods below too). Probably good to start making it aligned w/ how jsDoc wants it.